stl/inc/regex

```diff
@@ -1134,6 +1189,8 @@ _NODISCARD bool operator==(const match_results<_BidIt, _Alloc>& _Left, const mat
     }
 }
 
+// TRANSITION, VSO-900973
+// Overload resolution prefers implicit conversions over re-writes

```

Comment:
Is this really a problem for `match_results`, which doesn't inherit from a class type that implements comparison operators? I thought the issue with `sub_match` was that it implicitly converts to a `pair` specialization.

Reply:
I was being pessimistic without think too hard, but yeah I think you're right.